### ISSUE TITLE
Federation is not generating the good query with variables

### ISSUE THREAD

**Issue by hugues33** [apollographql/apollo-server#2890]
Hello,

I have a problem with the following query :
```
  query OFFER_QUERY($selectedOffer: String) {
    applicationsCodes
    permissions(applicationCodes: [$selectedOffer])
    applications(applicationCodes: [$selectedOffer]) {
      code
    }
```
when processed by Gateway, the query becomes :
```
query ($selectedOffer: String, $selectedOffer: String) {
  applicationsCodes
  permissions(applicationCodes: [$selectedOffer])
  applications(applicationCodes: [$selectedOffer]) {
    code
  }
```
which generates an error because the variable is defined twice.

@apollo/gateway: 0.6.6

Thanks in advance.

**Comment by hannesj** [comment]
Fixed by https://github.com/apollographql/apollo-server/pull/2840, which is included on 0.6.7.

**Comment by hugues33** [comment]
It's working fine now!

Thanks
